contributors/design-proposals/storage/dynamic_volume_limit.md

```diff
@@ -0,0 +1,106 @@
+# Dynamic attached volume limits
+
+## Goals
+
+Currently the number of volumes attachable to a node is either hardcoded or only configurable via an environment variable. Also
+existing limits only apply to well known volume types like EBS, GCE and is not available to all volume plugins.
+
+This proposal enables any volume plugin to specify those limits and also allows same volume type to have different volume
+limits depending on type of node.
+
+## Implementation Design
+
+### Prerequisite
+
+* This feature will be protected by an alpha feature gate, so as API and CLI changes needed for it. We are planning to call
+  the feature `DynamicVolumeThreshold`.
+
+### API Changes
+
+There is no API change needed for this feature. However existing `node.Status.Capacity` and `node.Status.Allocatable` will
+be extended to cover volume limits available on the node too.
+
+The key name that will store volume will be start with prefix `storage-`. So for example:
+
+* `storage-aws-ebs`
```

Comment:
Are there any restrictions on allowed characters?  Does this follow the existing resource naming convention?

Reply:
Yes it does. I will update the proposal:

```
// Resource names must be not more than 63 characters, consisting of upper- or lower-case alphanumeric characters,
// with the -, _, and . characters allowed anywhere, except the first or last character.
// The default convention, matching that for annotations, is to use lower-case names, with dashes, rather than
// camel case, separating compound words.
```